packages/ui/src/toast/toast.tsx

```diff
@@ -0,0 +1,289 @@
+import React, { useCallback, useMemo, useRef } from 'react';
+import ReachAlert from '@reach/alert';
+import { useRect } from '@reach/rect';
+import { useId } from '@reach/auto-id';
+
+import { Transition } from '../transition';
+import { Stack } from '../stack';
+import { Box, BoxProps } from '../box';
+import { Flex } from '../flex';
+import { Text } from '../text';
+import { ExclamationMarkCircleIcon, CheckmarkCircleIcon, CloseIcon } from '../icons';
+
+import { useTimeout } from '../hooks';
+
+import { ActionProps, ToastAction, ToastProps } from './types';
+import { useHover, useFocus } from 'use-events';
+import { transition } from '../theme/theme';
+
+const toneToIcon = {
+  critical: ExclamationMarkCircleIcon,
+  positive: CheckmarkCircleIcon,
+  none: () => null,
+};
+
+const Action = ({ label, onClick, removeToast, ...rest }: ActionProps) => {
+  const handleClick = useCallback(() => {
+    removeToast();
+    onClick();
+  }, [removeToast, onClick]);
+
+  return (
+    <Box
+      _hover={{
+        cursor: 'pointer',
+        textDecoration: 'underline',
+      }}
+      onClick={handleClick}
+      aria-hidden
+      {...rest}
+    >
+      <Text>{label}</Text>
+    </Box>
+  );
+};
+
+const getCustomProps = (props: ToastProps['toastProps'] = {}) => {
```

Comment:
Why does `ToastProps` have a property `toastProps`? 🤔

Reply:
It's kind of an escape hatch, when calling the toast, you can pass any custom styled-system props to the various elements of the toast, eg let's say I want to write a custom toast style for all my toasts:

```tsx
const makeCustomToast = (props: ToastProps) =>
  showToast({
    ...props,
    toastProps: { message: { fontSize: '24px', color: 'blue', fontWeight: 'bold' } },
  });
```

Reply:
@kyranjamie lmk if you have any thoughts/feedback on this, otherwise I will resolve this convo